usaon_benefit_tool/routes/assessment/node.py

```diff
@@ -0,0 +1,156 @@
+from typing import Literal
+
+import sqlalchemy
+from flask import Blueprint, Response, render_template, request, url_for
+from flask_login import login_required
+from flask_pydantic import validate
+from pydantic import BaseModel
+
+from usaon_benefit_tool import db
+from usaon_benefit_tool.forms import FORMS_BY_MODEL
+from usaon_benefit_tool.models.tables import AssessmentNode, Link, Node
+from usaon_benefit_tool.util.sankey import (
+    permitted_source_link_types,
+    permitted_target_link_types,
+)
+
+assessment_node_bp = Blueprint(
+    'node',
+    __name__,
+    url_prefix='/node/<int:node_id>',
+)
+Form = FORMS_BY_MODEL[AssessmentNode]
+
+
+@assessment_node_bp.route('/form', methods=['GET'])
+@login_required
+def form(assessment_id: int, node_id: int):
+    """View assessment node object."""
+    try:
+        assessment_node = _query_for_assessment_node(assessment_id, node_id)
+    except sqlalchemy.orm.exc.NoResultFound:
+        return Response(status=404)
+
+    form = Form(obj=assessment_node)
+
+    return render_template(
+        'assessment/_edit_node.html',
+        form=form,
+        assessment_id=assessment_id,
+        node=assessment_node.node,
+    )
+
+
+class _QueryModel(BaseModel):
+    direction: Literal["left", "right"]
+
+
+@assessment_node_bp.route('/form_new_link', methods=['GET'])
+@login_required
+@validate()
+def form_new_link(assessment_id: int, node_id: int, query: _QueryModel):
+    """Display a form to add a new link (to the left or right) of the node."""
+    try:
+        assessment_node = _query_for_assessment_node(assessment_id, node_id)
+    except sqlalchemy.orm.exc.NoResultFound:
+        return Response(status=404)
+
+    LinkForm = FORMS_BY_MODEL[Link]
+
+    # FIXME: I think this conditional is horrible!!!
```

Comment:
I'm weirdly OK leaving this as-is for now, as much as I hate it. I need time to think more and we need to get this in a test environment.